airsonic-main/src/main/java/org/airsonic/player/domain/PlayQueue.java

```diff
@@ -311,17 +311,7 @@ public synchronized void rearrange(int[] indexes) {
      * @param index The playlist index.
      */
     public synchronized void moveUp(int index) {
-        makeBackup();
-        if (index <= 0 || index >= size()) {
-            return;
-        }
```

Comment:
Just out of curiosity, any reason why the backup isn't made _after_ the noop condition?

Reply:
You're probably right, that would be better.  I just removed the duplicated code and preserved the original code path and am leaving refactoring/improvement to another PR.

Reply:
👍